core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -833,10 +850,12 @@ private[history] class AppListingListener(log: FileStatus, clock: Clock) extends
     var sparkUser: String = null
     var completed = false
     var appSparkVersion = ""
+    val jobToStatus = new mutable.HashMap[Int, String]
 
     def toView(): AttemptInfoWrapper = {
       val apiInfo = new v1.ApplicationAttemptInfo(
         attemptId,
+        applicationStatus.get,
```

Comment:
Here, the result of applicationStatus is always defined, so no need to return `Option[String]`.